src/pages/billing/Project.js

```diff
@@ -524,6 +527,43 @@ const ProjectDetail = ({ authorizeAndLoadAccounts, billingAccounts, billingProje
   const isGcpProject = billingProject.cloudPlatform === cloudProviders.gcp.label
 
   const error = Utils.maybeParseJSON(errorMessage)
+  const ErrorAlert = ({ error }) => {
```

Comment:
`ErrorAlert` should be defined outside of `ProjectDetail`. Components should not be defined inside other components. Because a new `ErrorAlert` function is created each time `ProjectDetail` is rendered, React can't tell that they are the same component. This results in `ErrorAlert` being unmounted and recreated on each render.

Reply:
Got it. That makes sense, thanks for explaining!